### PR TITLE
Add unit tests for burp.Methods

### PR DESCRIPTION
**Pull request by ThomasPerkins1123**
Hi,

I've analysed your code base and noticed that `burp.Methods` in the `knife` module is not fully tested.

I've written some tests that cover this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests should help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other particular classes that you consider important.

Whilst creating these tests I believe I noticed a small bug with your `do_modify_request` method which I have fixed for you.
Also your source files have been moved into `src/main` to more comfortably allow unit tests.

### CODE REVIEW

File: src/test/burp/MethodsTest.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import burp.Methods;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
```
Comment (Braavos6) [inline src/test/burp/MethodsTest.java:15]: Remove the extra line

File: src/test/burp/MethodsTest.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import javax.swing.*;
```
Comment (Braavos6) [inline src/test/burp/MethodsTest.java:16]: Try not to import wildcards

File: src/test/burp/MethodsTest.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+@RunWith(PowerMockRunner.class)
+public class MethodsTest {
+
+  @Rule public final ExpectedException thrown = ExpectedException.none();
+
+  @PrepareForTest({JOptionPane.class, Methods.class})
+  @Test
+  public void testPrompt_and_validate_input() {
+    PowerMockito.mockStatic(JOptionPane.class);
+    PowerMockito.when(JOptionPane.showInputDialog(anyString(), anyString())).thenReturn("").thenReturn("fooBar");
```
Comment (Braavos6) [inline src/test/burp/MethodsTest.java]: Line longer than 80 characters

File: src/test/burp/MethodsTest.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+  @Test
+  public void testEncodingException() throws UnsupportedEncodingException {
+    thrown.expect(NullPointerException.class);
+    Methods.encoding(null, 0, false);
+  }
+
+  @Test
+  public void testEncodingFalse() throws UnsupportedEncodingException {
+    Assert.assertArrayEquals("3\r\nfoo\r\n3\r\nBar\r\n0\r\n\r\n".getBytes(), Methods.encoding("fooBar".getBytes(), 3, false));
```
Comment (Braavos6) [inline src/test/burp/MethodsTest.java]: Line longer than 80 characters

File: src/test/burp/MethodsTest.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+  }
+
+  @Test
+  public void testDo_modify_request() {
+    Assert.assertArrayEquals("fooBaro".getBytes(), Methods.do_modify_request("foo".getBytes(), new int[]{3, 2}, "Bar".getBytes()));
+  }
+
+  @Test
+  public void testDo_modify_requestNull() {
+    Assert.assertArrayEquals("foo".getBytes(), Methods.do_modify_request("foo".getBytes(), new int[]{3, 2}, null));
```
Comment (Braavos6) [inline src/test/burp/MethodsTest.java]: Couldn't `testDo_modify_request` and `testDo_modify_requestNull` be squashed into one method as they are testing the same method and only a line long?

**Review by Braavos6 — CHANGES_REQUESTED** [review]
A few lines are longer than 80 characters. I didn't check each line but pretty sure there are some more there

**Review by paulkennethkent — CHANGES_REQUESTED** [review]
Splits this into multiple PRs (slack convo)

**Pull request closed without merging**